src/main/java/io/minio/MinioClient.java

```diff
@@ -152,7 +152,7 @@
    * @see #MinioClient(String endpoint, int port, String accessKey, String secretKey, boolean secure)
    */
   public MinioClient(String endpoint) throws InvalidEndpointException, InvalidPortException {
-    this(endpoint, 0, null, null, true);
+    this(endpoint, 0, null, null, endpoint==null?false:endpoint.startsWith("https://"));
```

Comment:
As ie mentioned in the comment, if endpoint is `localhost` or `myminio.com`, they are valid and should use HTTPS by default.
This check make it false.